### ISSUE TITLE
copy all tags and metadata block (eg. APPLICATION) with metaflac

### ISSUE THREAD

**Issue by mistepien** [xiph/flac#93]
It would be great to have an opportunity to copy ALL TAGS and metadata blocks from one flac file to the other without encoding.

**Comment by erikd** [comment]
Does the flac command line program not do that already? If it doesn't can you give me an example command and input file where all tags are not copied?

**Comment by mistepien** [comment]
no, flac makes even more. the point is to have full control over all tags WITHOUT encoding.

**Comment by erikd** [comment]
Sorry, I really have zero idea of what you are talking about. You need to explain it much better *and* provide an example.

**Comment by mistepien** [comment]
because copying tags and metadata block by metaflac means also ability to edit them. this is what for is metaflac. to copy, to remove, to edit. currently metaflac does not control all non-stream block.

**Comment by ktmf01** [comment]
With PR #549 copying of all metadata blocks (except seektable and streaminfo which can't and shouldn't be copied) can be done as follows

```
metaflac --remove --except-block-type=STREAMINFO,SEEKTABLE destinationfile.flac
metaflac --list --data-format=binary sourcefile.flac --except-block-type=STREAMINFO,SEEKTABLE | metaflac --append destinationfile.flac
```

This first removes the existing metadata blocks in the destination and then exports and imports metadata with two `metaflac` calls.

**Comment by ktmf01** [comment]
This has been implemented in #549
